### PR TITLE
Add ObjectSelectorEditor tests and fix a bug

### PR DESCRIPTION
**Pull request by hughbe**
Contributes to #817

### CODE REVIEW

**Comment by RussKie** [comment]
Could you please clarify what the bug was?

File: src/System.Windows.Forms.Design.Editors/src/System/ComponentModel/Design/ObjectSelectorEditor.cs [hunk 1]
```diff
@@ -42,10 +42,9 @@ public override object EditValue(ITypeDescriptorContext context, IServiceProvide
         {
             if (null != provider)
```
Comment (RussKie) [inline src/System.Windows.Forms.Design.Editors/src/System/ComponentModel/Design/ObjectSelectorEditor.cs]: Please invert if to reduce nesting

File: src/System.Windows.Forms.Design.Editors/src/System/ComponentModel/Design/ObjectSelectorEditor.cs [hunk 2]
```diff
@@ -42,10 +42,9 @@ public override object EditValue(ITypeDescriptorContext context, IServiceProvide
...
             {
-                IWindowsFormsEditorService edSvc = (IWindowsFormsEditorService)provider.GetService(typeof(IWindowsFormsEditorService));
-                if (edSvc != null)
```
Comment (RussKie) [inline src/System.Windows.Forms.Design.Editors/src/System/ComponentModel/Design/ObjectSelectorEditor.cs]: ditto, and the next line can be reverted

**Comment by hughbe** [comment]
> Can you please clarify what the bug was?

If you called `new ObjectSelectorEditor().EditValue(null, provider, new object())` then we'd throw an `InvalidCastException`

**Pull request merged**